packages/x-charts/src/context/CartesianProvider/CartesianProviderPro.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { useDrawingArea } from '../../hooks/useDrawingArea';
+import { useSeries } from '../../hooks/useSeries';
+import { CartesianContext } from './CartesianContext';
+import { normalizeAxis } from './normalizeAxis';
+import { computeValue } from './computeValue';
+import { ExtremumGettersConfig } from '../../models';
+import { useZoom } from '../ZoomProvider/useZoom';
+import { CartesianContextProviderProps } from './CartesianProvider';
+
+function CartesianContextProviderPro(props: CartesianContextProviderProps) {
+  const prevContext = React.useContext(CartesianContext);
+
+  if (prevContext.isInitialized) {
+    return props.children;
+  }
+
+  return <CartesianContextProviderReal {...props} />;
+}
+
+const calculateZoom = (
+  data: [number | null, number | null],
+  zoom: [number, number],
+): [number | null, number | null] => {
+  const [min, max] = data;
+  const [minZoom, maxZoom] = zoom;
+
+  if (min === null || max === null || !Number.isFinite(min) || !Number.isFinite(max)) {
+    return [null, null];
+  }
+  const diff = max - min;
+  const newMin = min + (diff * minZoom) / 100;
+  const newMax = min + (diff * maxZoom) / 100;
+
```

Comment:
This will probably require a specific behavior for bar chart

Reply:
Yeah, I'll need to create a lot of logic for the different types 👍